src/core/control/xojfile/XmlParserHelper.h

```diff
@@ -0,0 +1,280 @@
+ */
+template <typename T>
+T getAttribMandatory(std::u8string_view name, const AttributeMap& attributeMap, const T& defaultValue = {},
+                     bool warn = true);
+// specializations
+template <>
+std::optional<const char*> getAttrib<const char*>(std::u8string_view name, const AttributeMap& attributeMap);
+template <>
+std::optional<c_string_utf8_view> getAttrib<c_string_utf8_view>(std::u8string_view name,
+                                                                const AttributeMap& attributeMap);
+template <>
+std::optional<string_utf8_view> getAttrib<string_utf8_view>(std::u8string_view name, const AttributeMap& attributeMap);
+template <>
+std::optional<fs::path> getAttrib<fs::path>(std::u8string_view name, const AttributeMap& attributeMap);
+template <>
+std::optional<LineStyle> getAttrib<LineStyle>(std::u8string_view name, const AttributeMap& attributeMap);
+
+// "color" attribute
+Color getAttribColorMandatory(const AttributeMap& attributeMap, const Color& defaultValue, bool bg = false);
+// Attempt to match string with background-specific color "translations"
+std::optional<Color> parseBgColor(string_utf8_view sv);
+// Parse str as a RGBA hex color code
+std::optional<Color> parseColorCode(std::string_view sv);
+// Attempt to match string with predefined color names
+std::optional<Color> parsePredefinedColor(string_utf8_view sv);
+
+// Decode C-string of Base64 encoded data into a string of binary data
+std::string decodeBase64(std::string_view base64data);
+
+namespace detail {
+
+// SFINAE logic for checking named enums and utf8 views
+template <typename, typename = void>
+inline constexpr bool has_names_v = false;
+
+template <typename T>
+inline constexpr bool has_names_v<T, std::void_t<decltype(T::NAMES)>> = true;
+
+
+template <typename, typename = void>
+inline constexpr bool has_value_enum_v = false;
+
+template <typename T>
+inline constexpr bool has_value_enum_v<T, std::void_t<typename T::Value>> = std::is_enum_v<typename T::Value>;
+
+
+template <typename>
+inline constexpr bool is_utf8_view_v = false;
+
+template <typename It, typename Sen>
+inline constexpr bool is_utf8_view_v<xoj::util::utf8_view<It, Sen>> = true;
+
+
+template <typename T>
+constexpr bool always_false = false;
+
+// Check for floating-point std::from_chars support
+#ifndef HAS_FLOAT_FROM_CHARS
+#if defined(_GLIBCXX_RELEASE)
+// libstdc++ supports floating-point from_chars since GCC 11
+#define HAS_FLOAT_FROM_CHARS (_GLIBCXX_RELEASE >= 11)
+#elif defined(_LIBCPP_VERSION)
+// libc++ supports floating-point from chars since version 20.1.0
+#define HAS_FLOAT_FROM_CHARS (_LIBCPP_VERSION >= 200100)
+#elif defined(_MSC_VER)
+// MSVC STL supports float from_chars since VS 2017 15.7
+#define HAS_FLOAT_FROM_CHARS (_MSC_VER >= 1914)
+#else
+// Unknown standard library
+#define HAS_FLOAT_FROM_CHARS 0
+#endif
+#endif  // HAS_FLOAT_FROM_CHARS
```

Comment:
Rather than listing the possible implementations of std, it may be cleaner to have a `check_cxx_source_compiles` in some cmake file.
Actually, I think it is better to raise an error if the compiler does not support `from_chars`, and add a cmake flag to disable the feature.
IMO, the person compiling should be explicitly made aware of the performance loss, and knowingly choose to do so.

Btw, do some of our CI targets not support it?